scripts/plotting/tape_recorder.py

```diff
@@ -0,0 +1,543 @@
+    mymap = mcolors.LinearSegmentedColormap.from_list('my_colormap', colors)
+  
+    return mymap
+
+
+
+def precip_cmap(n, nowhite=False):
+    """ combine two existing color maps to create a diverging color map with white in the middle.
+    browns for negative, blues for positive
+    n = the number of contour intervals
+    """
+    if (int(n/2) == n/2):
+        # even number of contours
+        nwhite=1
+        nneg=n/2
+        npos=n/2
+    else:
+        nwhite=2
+        nneg = (n-1)/2
+        npos = (n-1)/2
+
+    if (nowhite):
+        nwhite=0
+
+    colors1 = plt.cm.YlOrBr_r(np.linspace(0,1, int(nneg)))
+    colors2 = plt.cm.GnBu(np.linspace(0,1, int(npos)))
+    colorsw = np.ones((nwhite,4))
+
+    colors = np.vstack((colors1, colorsw, colors2))
+    mymap = mcolors.LinearSegmentedColormap.from_list('my_colormap', colors)
+
+    return mymap
+
+def precip_cmap_nowhite(n):
+    """ combine two existing color maps to create a diverging color map with white in the middle.
+    browns for negative, blues for positive
+    n = the number of contour intervals
+    """
+    if (int(n/2) == n/2):
+        # even number of contours
+        nneg=n/2
+        npos=n/2
+    else:
+        nneg = (n-1)/2
+        npos = (n-1)/2
+
+    colors1 = plt.cm.YlOrBr_r(np.linspace(0,0.8, int(nneg)))
+    colors2 = plt.cm.GnBu(np.linspace(0.2,1, int(npos)))
+
+    colors = np.vstack((colors1, colors2))
+    mymap = mcolors.LinearSegmentedColormap.from_list('my_colormap', colors)
+
+    return mymap
+
+
+
+def fixcesmtime(dat,timebndsvar='time_bnds'):
+    """ Fix the CESM timestamp using the average of time_bnds"""
+
+    try:
+        timebndavg = np.array(dat.isel(M=0)[timebndsvar],
+                  dtype='datetime64[s]').view('i8').mean(axis=1).astype('datetime64[s]')
+        dat['time'] = timebndavg
+    except:
+        timebndavg = np.array(dat[timebndsvar],
+                  dtype='datetime64[s]').view('i8').mean(axis=1).astype('datetime64[s]')
+        dat['time'] = timebndavg
+
+    return dat
+
+
+def get5by5coords_zmplots():
+    """ positioning for 5x5 plots """
+    x1 = [0.02,0.225,0.43,0.635,0.84,
+          0.02,0.225,0.43,0.635,0.84,
+          0.02,0.225,0.43,0.635,0.84,
+          0.02,0.225,0.43,0.635,0.84,
+          0.02,0.225,0.43,0.635,0.84] 
+    x2 = [0.18,0.385,0.59,0.795,1,
+          0.18,0.385,0.59,0.795,1,
+          0.18,0.385,0.59,0.795,1,
+          0.18,0.385,0.59,0.795,1,
+          0.18,0.385,0.59,0.795,1]
+    y1 = [0.8,0.8,0.8,0.8,0.8,
+          0.6,0.6,0.6,0.6,0.6,
+          0.4,0.4,0.4,0.4,0.4,
+          0.2,0.2,0.2,0.2,0.2,
+          0.,0.,0.,0.,0.]
+    y2 = [0.95,0.95,0.95,0.95,0.95,
+          0.75,0.75,0.75,0.75,0.75,
+          0.55,0.55,0.55,0.55,0.55,
+          0.35,0.35,0.35,0.35,0.35,
+          0.15,0.15,0.15,0.15,0.15]
+    
+    return x1, x2, y1, y2
+
+
+
+import matplotlib.pyplot as plt
+import matplotlib as mpl
+
+import importlib
+
+import numpy as np
```

Comment:
I don't think any of this is needed, except possibly the `mpl` import, although I think much of that can be replaced with the use of the other `matplotlib` import calls done above.  If you find that it needs to stay then I would still move it to the top of the file so that is grouped together with the rest of the `import` statements as well.

Reply:
Done.